### ISSUE TITLE
Remove confusing implicit conversions on GeoTiff

### ISSUE THREAD

**Issue by metasim** [geotrellis/geotrellis#1780]
Invoking `(t: MultibandGeoTiff).convert(celltype)` results in an [implicit conversion](https://github.com/geotrellis/geotrellis/blob/master/raster/src/main/scala/geotrellis/raster/io/geotiff/MultibandGeoTiff.scala#L121) to `MultibandTile` and a call to its `convert` method. This results in somewhat confusing semantics, where the GeoTiff metadata is lost and the return type is different from what you started with. Two suggestions:

1. add `def convert(cellType: CellType) = this.copy(tile = tile.convert(cellType))` to `MultibandGeoTiff`.
2. Consider removing the implicit conversion linked above, and replacing whatever features it breaks with a type class or an implicit enrichment class.

Associated Gitter dialog: https://gitter.im/geotrellis/geotrellis?at=58208b93e462097a301bb613

**Comment by lossyrob** [comment]
To repeat my gitter comment:

```
perhaps taking that implicit out is the fix
we have a mapTile, so you could simply do geoTiff.mapTile(_.convert(…))
```

**Comment by metasim** [comment]
I agree: Basically just implement 2 above. The title should probably be changed.

**Comment by lossyrob** [comment]
Renamed, fixed w/ #1782
